StripeUICore/StripeUICore/Source/Views/DynamicImageView.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  DynamicImageView.swift
+//  StripeUICore
+//
+//  Created by Eduardo Urias on 11/10/22.
+//  Copyright © 2022 Stripe, Inc. All rights reserved.
+//
+
+import UIKit
+
+/// A `UIImageView` that dynamically changes it's `image` according to the brightness of the
+/// `pairedColor`.
+@objc(STP_Internal_DynamicImageView)
+@_spi(STP) public class DynamicImageView: UIImageView {
+    private let pairedColor: UIColor
+    private let darkImage: UIImage?
+    private let lightImage: UIImage?
+
+    private var currentImage: UIImage? {
+        pairedColor.resolvedColor(with: traitCollection).isDark
+            ? darkImage ?? lightImage
+            : lightImage ?? darkImage
+    }
```

Comment:
Super nit, feel free to drop - feels confusing that current image is not, in fact, the currently displayed image of UIImageView - that would be `image`. 

IMO this is slightly better as a `func makeImage(for traitCollection) -> UIImage` or something.

Reply:
makes sense, changed